src/llvm/function.cr

```diff
@@ -3,6 +3,8 @@ require "./value_methods"
 struct LLVM::Function
   include LLVM::ValueMethods
 
+  property debug_params : Array(LibLLVMExt::Metadata) = [] of LibLLVMExt::Metadata
+
```

Comment:
I don't think this belongs here.

We should keep a `Hash(LLVM::Function, Array(LibLLVMExt::Metadata))` in `Codegen`.

The LLVM library is meant to also be used by other users, not exclusive to Crystal, so we shouldn't add Crystal-specific stuff here.

Reply:
I had to keep it here as it sits in context.fun that is available from any place. I am open for your suggestions how to refactor it.

Reply:
Here's the suggestion:

> We should keep a Hash(LLVM::Function, Array(LibLLVMExt::Metadata)) in Codegen.

Reply:
Ok. I will check if I can move it there without any possible breaking of the code.
Will do my best.

Reply:
Sure! If you have trouble let me know and I can try doing it, if you provide a test case to debug it and see if it still works fine.